### PR TITLE
Case export migration

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @NoahCarnahan couple things going on here. best read 🐟. this adds the ability to revert a migration. it gets all current app ids if none are specified (this is important for migrating case exports). `last_app_versions` was completely broken. fixed it and added a test (it's not being used anywhere which is why we didn't notice, but it was confusing when i was trying to inspect an export)

cc: @dannyroberts

### CODE REVIEW

File: corehq/apps/export/models/new.py [hunk 1]
```diff
@@ -954,7 +959,7 @@ def generate_schema_from_builds(cls, domain, app_id, identifier, force_rebuild=F
                 identifier,
             )
 
-            current_schema.record_update(app_id, app.version)
+            current_schema.record_update(app.copy_of or app_id, app.version)
```
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: for case exports this was completely busted since we'd look at apps in all domains not just the one passed in

**Comment by NoahCarnahan** [comment]
Looks good to me 👍

**Comment by czue** [comment]
build fail looks real
```
======================================================================
ERROR: test_create_form_export (corehq.apps.export.tests.test_export_views.ExportViewTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/export/tests/test_export_views.py", line 48, in test_create_form_export
    {'export_tag': 'my_sweet_xmlns'}
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 470, in get
    **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 286, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 358, in generic
    return self.request(**r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 440, in request
    six.reraise(*exc_info)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/vendor/lib/python2.7/site-packages/django/views/generic/base.py", line 69, in view
    return self.dispatch(request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/style/decorators.py", line 303, in _wrapped
    return view_func(class_based_view, request, *args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 29, in _wrapper
    return bound_func(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/domain/decorators.py", line 101, in _inner
    return view_func(req, domain_name, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/users/decorators.py", line 21, in _inner
    return view_func(request, domain, *args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 25, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/mnt/commcare-hq-ro/corehq/apps/export/views.py", line 1476, in dispatch
    return super(BaseModifyNewCustomView, self).dispatch(request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/style/decorators.py", line 89, in _wrapped
    return view_func(class_based_view, request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/export/views.py", line 1436, in dispatch
    return super(BaseNewExportView, self).dispatch(request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/style/decorators.py", line 89, in _wrapped
    return view_func(class_based_view, request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/export/views.py", line 154, in dispatch
    return super(BaseExportView, self).dispatch(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 29, in _wrapper
    return bound_func(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/domain/decorators.py", line 101, in _inner
    return view_func(req, domain_name, *args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 25, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/mnt/commcare-hq-ro/corehq/apps/domain/views.py", line 215, in dispatch
    return super(LoginAndDomainMixin, self).dispatch(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/views/generic/base.py", line 87, in dispatch
    return handler(request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/export/views.py", line 1491, in get
    xmlns,
  File "/mnt/commcare-hq-ro/corehq/apps/export/models/new.py", line 948, in generate_schema_from_builds
    app_build_ids.extend(cls._get_current_app_ids_for_domain(domain))
  File "/mnt/commcare-hq-ro/corehq/apps/export/models/new.py", line 1083, in _get_current_app_ids_for_domain
    raise BadExportConfiguration('Form exports should only use one app_id and this should not be called')
BadExportConfiguration: Form exports should only use one app_id and this should not be called
```

**Comment by czue** [comment]
looks good to me too when :white_check_mark:

**Pull request merged**